### PR TITLE
feat(docker): Rename images to new defaults

### PR DESCRIPTION
**Pull request by heliocastro**
Rename Docker image 'ort' to 'ort-minimal' and  'ort-extended' to 'ort' as agreed in community vote
to make it easier for new users to understand difference between ORT Docker images.
Additionally the language component images now have named version with a hash to enable checking if contents
of an image has changed without changing core language.

Closes: #7798

### CODE REVIEW

File: .github/actions/ortdocker/check_image.py
Comment (fviernau) [inline .github/actions/ortdocker/check_image.py:31]: Is this hashing supposed to solve https://github.com/oss-review-toolkit/ort/issues/7798 ?

How does it work with changes which do not affect build args? 
e.g. I've added `git-lfs` a while ago, but in the published image it's not available for unknown reason. If I build image locally it works as expected.

**Review by sschuberth — CHANGES_REQUESTED** [review]

File: .github/actions/ortdocker/check_image.py
Comment (heliocastro) [reply .github/actions/ortdocker/check_image.py:31]: Yes, this supposes to solve this specific bug.
When build args not changes, hash will be the same, so will check if the image tagged in ghcr.io with {version}-{hash} exists. If exists, not build, since is the same.

File: .github/actions/ortdocker/action.yml
Comment (sschuberth) [reply .github/actions/ortdocker/action.yml:23]: All these lines change indentation for no reason / other code changes:

image

File: .github/actions/ortdocker/action.yml
Comment (sschuberth) [inline .github/actions/ortdocker/action.yml:47]: Please also don't mix-in other unrelated changes like this changing of quotes.

File: .github/actions/ortdocker/check_image.py
Comment (sschuberth) [inline .github/actions/ortdocker/check_image.py:55]: Is this change intended, and does it belong to the topic of this commit?

**Review by sschuberth — CHANGES_REQUESTED** [review]
> Additionally the language component images now have named version with a hash to enable checking if contents of an image has changed without changing core language.

I propose to do this change in a preceding commit as it's not really on the topic of just renaming the Docker images.

File: .github/actions/ortdocker/check_image.py
Comment (sschuberth) [reply .github/actions/ortdocker/check_image.py:55]: Please explain this here instead of just marking my comment as "resolved".

Comment (heliocastro) [reply .github/actions/ortdocker/check_image.py:55]: Did i not explain one line above ?

File: .github/actions/ortdocker/action.yml
Comment (sschuberth) [reply .github/actions/ortdocker/action.yml]: > or change all to double quote, or to single quote

I have no problem with that, but this should be done in a separate commit consistently for all files then.

File: .github/actions/ortdocker/check_image.py
Comment (fviernau) [reply .github/actions/ortdocker/check_image.py:31]: I don't understand yet why 'git-lfs' is not in the docker built by CI, see [1].
That commit [1] is not changing build args, but I wonder if your change here fixes that problem implicitly, e.g. so that docker layer caching starts detecting the modified `apt-get install` e.g.

Can you help me understanding this bit?

[1] https://github.com/oss-review-toolkit/ort/commit/a4b12df0acc6329cc164712e189d2c865c4280b2

Comment (heliocastro) [reply .github/actions/ortdocker/check_image.py:31]: Images suppose to be immutable, .
The check happens in this way:
current: base-17-jdk-jammy,  so the version is 17-jdk-jammy.

The test before was checking for version, so every time found a container in ghcr with the version that matches, simply pass and not build it. 

The problem is that we changed the internal content without change the main version. 
So when you added git-lfs, even changing inside, the new container was never generated.

Noe we have a hash based on the arguments as part of the version, so change the build arguments will declare as a new image and will build and push as new.

And the main registry will not receive pushes from content during pull_requests. Only in nighlty or in the tag events.
After this PR is approved, we trigger a workflow dispatch event and the issue will be gone,

File: .github/actions/ortdocker/action.yml
Comment (heliocastro) [reply .github/actions/ortdocker/action.yml]: The name change and the fix are coupled, do it in two steps will not benefit at all. So the pr is done, just waiting for review.

Comment (sschuberth) [reply .github/actions/ortdocker/action.yml]: > The name change and the fix are coupled

Changing the quoting style is not coupled to renaming images.

Comment (heliocastro) [reply .github/actions/ortdocker/action.yml]: No, the hashing change not the quotes. quotes will be done other time as you suggested.

Comment (heliocastro) [reply .github/actions/ortdocker/action.yml]: Is weird that this is threaded over the hashing changes. Sorry for the confusion.

File: .github/actions/ortdocker/check_image.py
Comment (fviernau) [reply .github/actions/ortdocker/check_image.py:31]: Thanks!

**Comment by heliocastro** [comment]
> > Additionally the language component images now have named version with a hash to enable checking if contents of an image has changed without changing core language.
> 
> I propose to do this change in a preceding commit as it's not really on the topic of just renaming the Docker images.

Is mention before, the change and fix are coupled, no reason to do two separate entries.
Can we move forward now ?

**Review by tsteenbe — DISMISSED** [review]

**Review by tsteenbe — DISMISSED** [review]

**Review by fviernau — DISMISSED** [review]

File: Dockerfile
Comment (heliocastro) [reply Dockerfile:436]: Because all builds relies on minimal target.  If you not do that, will break the build of all components. Is all about context usage.
Will not break the build, will just make it huge slow, as every sngle component will buils docker from scratch, like no cached exists

File: scripts/custom_docker.sh
Comment (heliocastro) [reply scripts/custom_docker.sh:33]: Do we really need to be so nitpicking ? At some point i had people that complained to me to have so much commits in the pull request.

File: scripts/custom_docker.sh
Comment (heliocastro) [reply scripts/custom_docker.sh:30]: Maybe, but this should go for a different  PR, i won't touch this right now

File: .github/workflows/docker-ort.yml
Comment (heliocastro) [reply .github/workflows/docker-ort.yml:165]: No, it can't, otherwise we end up with broken images again, because the chain of dependency is broken.

File: .github/actions/ortdocker/action.yml
Comment (heliocastro) [reply .github/actions/ortdocker/action.yml:57]: Changed the message.

File: scripts/custom_docker.sh
Comment (sschuberth) [reply scripts/custom_docker.sh:33]: Well, the rule is simple: Everything that is not absolutely *required* to get the topic of the commit implemented should be moved to a separate commit with its own rationale. On the other hand, intermediate code states during development and fixups for previous commits in the same PR should be consolidated.

Comment (heliocastro) [reply scripts/custom_docker.sh:33]: Moved to a new commit

File: scripts/docker_build.sh
Comment (heliocastro) [reply scripts/docker_build.sh:54]: So first there are complain that i made a big commit with al parts that change, and i should split for better understand.
Now i should squash all commits again because there are too many commits and mostly return to the initial situation ?

**Review by tsteenbe — APPROVED** [review]

**Review by tsteenbe — APPROVED** [review]

**Pull request merged**
